pypesto/engine/mpi_pool.py

```diff
@@ -29,14 +31,19 @@ class MPIPoolEngine(Engine):
     def __init__(self):
         super().__init__()
 
-    def execute(self, tasks: List[Task]):
+    def execute(self, tasks: List[Task], progress_bar: bool = True):
         """Pickle tasks and distribute work to workers."""
 
         pickled_tasks = [pickle.dumps(task) for task in tasks]
 
         n_procs = MPI.COMM_WORLD.Get_size()   # Size of communicator
         logger.info(f"Performing parallel task execution on {n_procs-1} "
                     f"workers with one manager.")
+
         with MPIPoolExecutor() as executor:
-            results = executor.map(work, pickled_tasks)
+            results = executor.map(work,
+                                   tqdm(pickled_tasks,
+                                        total=len(tasks),
```

Comment:
```suggestion
```
Can be left out as if it is unspecified it will always try to use `len(pickled_tasks)`. In case that would not be possible it could change to `floor('inf')` or something else while if you specify it it would then result in an error.

Reply:
I am a bit confused, the `total` key word tells the progress bar, how many steps the bar should take... every time a new object is returned by the iterable, the progress bar is then updated. Am I missing anything regarding the `tasks` vs `pickled_tasks` here?

Reply:
in this case, the pickled_tasks (and the tasks) have a `__len__` attribute, therefore the total argument should indeed be unnecessary (to be tested I guess).

Reply:
> total: int or float, optional
>The number of expected iterations. If unspecified, len(iterable) is used if possible. If float("inf") or as a last resort, only basic >progress statistics are displayed (no ETA, no progressbar). If gui is True and this parameter needs subsequent updating, >specify an initial arbitrary large positive number, e.g. 9e9.

from https://tqdm.github.io/docs/tqdm/ in `__init__`

Reply:
I know, what the keyword does, I just overlooked the `len(pickled_task)` option..